### ISSUE TITLE
PostgreSQL breaking changes for enums and JSON not shown

### ISSUE THREAD

**Issue by nikolasburk** [prisma/upgrade#80]
I'm currently going through the upgrade process but the SQL statements to convert `TEXT` to `enum` and `json` types are not shown.

Here's my Prisma 1 datamodel:

```graphql
type User {
  id: ID! @id
  email: String @unique
  name: String!
  role: Role! @default(value: CUSTOMER)
  jsonData: Json 
  profile: Profile
  posts: [Post!]!
}
type Post {
  id: ID! @id
  createdAt: DateTime! @createdAt
  updatedAt: DateTime! @updatedAt
  title: String!
  content: String
  published: Boolean! @default(value: false)
  author: User @relation(link: TABLE)
  categories: [Category!]!
}
type Profile {
  id: ID! @id
  bio: String
  user: User! @relation(link: INLINE)
}
type Category {
  id: ID! @id
  name: String!
  posts: [Post!]!
}
enum Role {
  ADMIN
  CUSTOMER
}
```

I used the Upgrade CLI successfully to:

- set `DEFAULT` constraints
- replicate `@createdAt` behavior in Prisma 2
- fix 1-1 relations by adding `UNIQUE` constraints
- migrate IDs from `varchar(25)` to `varchar(30)`

However, when moving to the next step with the breaking changes, the Upgrade CLI only shows this output:

```
➤ Adjust your database schema (these changes break Prisma 1)
Run the following SQL statements against your database:

  Fix one-to-many table relations
  https://pris.ly/d/schema-incompatibilities#all-non-inline-relations-are-recognized-as-m-n

    ALTER TABLE "dr10$dev"."Post" ADD COLUMN "authorId" CHARACTER VARYING(25);
    UPDATE "dr10$dev"."Post" SET "authorId" = "dr10$dev"."_PostToUser"."B" FROM "dr10$dev"."_PostToUser" WHERE "dr10$dev"."_PostToUser"."A" = "dr10$dev"."Post"."id";
    ALTER TABLE "dr10$dev"."Post" ADD FOREIGN KEY ("authorId") REFERENCES "dr10$dev"."User"("id");
    DROP TABLE "dr10$dev"."_PostToUser";



➤ Next Steps

After you executed one or more of the previous SQL statements against your database,
please run the following two commands to refresh your Prisma 2 schema and check
the changes.

  1. Run `npx prisma introspect` again to refresh your Prisma 2 schema.
  2. Run `npx prisma-upgrade` again.

If you can't or don't want to execute the remaining SQL statements right now, you can
skip to the last step where the Upgrade CLI adds missing attributes to your Prisma 2
schema that are not picked up by introspection.
```

**Comment by pablosat** [comment]
Hello, I am in the same process of upgrading from prisma1 to prisma2 and I had the same problem... but first I want to know if once you have executed all SQL (not breaking) statements and then introspected and upgraded your DB, there were some SQL statements showed again (as if you hadn't executed them).

Anyway... what I did to show the ENUM and JSON breaking statements, I did some changes to the file `node_modules/prisma-upgrade/dist/cli/index.js`: after the `MigrateScalarListOp` case in line 432, add the following cases
```javascript
        case "SetJsonTypeOp":
          console.log(`  ${kleur.bold(`Fix columns with JSON data types`)}`);
          console.log(
            `  ${kleur.gray(`https://pris.ly/d/schema-incompatibilities#json-type-is-represented-as-text-in-database`)}`
          );
          console.log();
          console.log(redent(queries.join("\n"), 4));
          console.log();
          console.log();
          break
        case "SetEnumTypeOp":
          console.log(`  ${kleur.bold(`Fix columns with ENUM data types`)}`);
          console.log(
            `  ${kleur.gray(`https://pris.ly/d/schema-incompatibilities#enums-are-represented-as-text-in-database`)}`
          );
          console.log();
          console.log(redent(queries.join("\n"), 4));
          console.log();
          console.log();
          break
```
Once you have added these lines, start the upgrade process, and when the not breaking SQL statements are shown for first time, execute all of them less the queries for default values for enum fields (in your case role CUSTOMER). Then introspect/upgrade again, and see the breaking changes (now you will see ENUM and JSON statements) and execute them. Later, introspect/upgrade again and run the remaining not breaking SQL statements for the default value for enums. Finally instrospect/upgrade for the last time and now your datamodel will have the enums and jsons.

I hope this help you and sorry for my english

**Comment by matthewmueller** [comment]
This is fixed in `0.0.41`. Sorry about thatr!
